js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/Sidebar.tsx

```diff
@@ -80,7 +81,20 @@ export const AssetGraphExplorerSidebar = React.memo(
         }
       }
     };
-    const [openNodes, setOpenNodes] = React.useState<Set<string>>(new Set());
+    const [openNodes, setOpenNodes] = useQueryAndLocalStoragePersistedState<Set<string>>({
+      // include pathname so that theres separate storage entries for graphs at different URLs
+      // eg. independent group graph should persist open nodes separately
+      localStorageKey: `asset-graph-open-nodes-${
+        location.pathname.split(explorerPathSeparator)[0]
```

Comment:
Ahhh man I see why you did it this way, and I can't think of a better way that isn't 10+ more lines of code... could we move this to PipelinePathUtils and give it an appropriate name so that it's co-located with the other path stuff and we're more likely to notice it if/when we change the path structure? I'm worried this is just subtle enough that we'll break it and it'll stay broken for many months haha

Reply:
Edit - actually, this component already has the explorerPath as a parsed prop -- could we more explicitly list out  `${isGlobalGraph}-${explorerPath.pipelineName}`? (I think that global / not global and the group / job name should be all that is encoded in the URL fragment currently?)

Reply:
Ah good call